base/src/pretty_print.rs

```diff
@@ -264,14 +264,52 @@ impl<'a> ExprPrinter<'a> {
     where
         Id: AsRef<str>,
     {
-        self.pretty_expr_(BytePos::from(0), expr).append(
+        self.pretty_expr_with_shebang_line(BytePos::from(0), expr).append(
             self.comments(Span::new(
                 expr.span.end,
                 BytePos::from(self.source.src().len()),
             )),
         )
     }
 
+    fn find_shebang_line(&'a self) -> Option<&str> {
+        let src = self.source.src();
+        let len = src.len();
+        if len > 1 && &src[0..2] == "#!" {
+            let newline_index = src.find('\n');
```

Comment:
Instead of find + indexing it is possible to use `src.lines().next()` to get the first line.

https://doc.rust-lang.org/std/primitive.str.html#method.lines